wheatley/row_generation/complib_composition_generator.py

```diff
@@ -75,22 +75,38 @@ def __init__(self, comp_id: int, access_key: Optional[str]=None) -> None:
         request_rows.raise_for_status()
         # Parse the request responses as JSON
         response_rows = json.loads(request_rows.text)
-
-        # Derive the rows, calls and stage from the JSON response
-        self.loaded_rows: List[Tuple[Row, Optional[str]]] = [
+        unparsed_rows = response_rows['rows']
+        # Determine the start row of the composition
+        num_starting_rounds = 0
+        while unparsed_rows[num_starting_rounds][0] == unparsed_rows[0][0]:
+            num_starting_rounds += 1
+        self._start_stroke = Stroke.from_index(num_starting_rounds)
+
+        # Unpack the raw rows and calls from the JSON
+        unshifted_rows_and_calls: List[Tuple[Row, List[str]]] = [
             (
                 Row([Bell.from_str(bell) for bell in row]),
-                None if call == '' else call
-            ) for row, call, property_bitmap in response_rows['rows'][2:]
+                [] if call == '' else call.split(';')
+            ) for row, call, property_bitmap in unparsed_rows
+        ]
+
+        # CompLib's API returns calls on the row which they are *called*, but Wheatley expects them on the
+        # row they *take effect*.  So we push all the calls along by two rows, ignoring the fact that
+        # `That's All; Stand` falls off the end.
```

Comment:
This comment has lost me a bit. Which part expects them on the row they take effect? 
My impression was this generator is generating calls to make at the start of a row, so should return them at the place they are called?

Reply:
This is another thing that I was half way through writing.  I'm going to put a top-level post in this PR explaining what's going on.